### ISSUE TITLE
[Extensibility] "invalid URL, scheme is not http"

### ISSUE THREAD

**Issue by DazWilkin** [deislabs/akri#77]
**Describe the bug**

**Output of `kubectl get pods,akrii,akric -o wide`**

```bash
kubectl apply --filename=nessie.yaml
configuration.akri.sh/nessie created
```
And:
```bash
kubectl get all
NAME                                              READY   STATUS    RESTARTS   AGE
pod/akri-agent-daemonset-6gjbr                    1/1     Running   0          89s
pod/akri-controller-deployment-848858df4b-cgplb   1/1     Running   0          89s

NAME                 TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE
service/kubernetes   ClusterIP   10.152.183.1   <none>        443/TCP   26h

NAME                                  DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
daemonset.apps/akri-agent-daemonset   1         1         1       1            1           kubernetes.io/os=linux   89s

NAME                                         READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/akri-controller-deployment   1/1     1            1           89s

NAME                                                    DESIRED   CURRENT   READY   AGE
replicaset.apps/akri-controller-deployment-848858df4b   1         1         1       89s
```

**Kubernetes Version: [e.g. Native Kubernetes 1.19, MicroK8s 1.19, Minikube 1.19, K3s]**

**Expected behavior**

**Logs (please share snips of applicable logs)**

```bash
kubectl logs pods/akri-agent-daemonset-6gjbr
akri.sh Agent start
akri.sh KUBERNETES_PORT found ... env_logger::init
[2020-10-27T19:57:34Z TRACE agent] akri.sh KUBERNETES_PORT found ... env_logger::init finished
[2020-10-27T19:57:34Z TRACE agent::util::slot_reconciliation] periodic_slot_reconciliation - start
[2020-10-27T19:57:34Z TRACE akri_shared::k8s] Loading in-cluster config
[2020-10-27T19:57:34Z INFO  agent::util::config_action] do_config_watch - enter
[2020-10-27T19:57:34Z TRACE akri_shared::k8s] Loading in-cluster config
[2020-10-27T19:57:34Z TRACE agent::util::slot_reconciliation] periodic_slot_reconciliation - iteration pre delay_for
[2020-10-27T19:57:34Z TRACE akri_shared::akri::configuration] get_configurations enter
[2020-10-27T19:57:34Z TRACE akri_shared::akri::configuration] get_configurations kube_client.request::<KubeAkriInstanceList>(akri_config_type.list(...)?).await?
[2020-10-27T19:57:34Z TRACE akri_shared::akri::configuration] get_configurations return
[2020-10-27T19:57:34Z TRACE agent::util::config_action] watch_for_config_changes - start
[2020-10-27T19:57:44Z TRACE agent::util::slot_reconciliation] periodic_slot_reconciliation - iteration call reconiler.reconcile
[2020-10-27T19:57:44Z TRACE agent::util::slot_reconciliation] reconcile - thread iteration start [Mutex { data: {} }]
[2020-10-27T19:57:44Z TRACE agent::util::slot_reconciliation] get_node_slots - Command failed to call crictl: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }
[2020-10-27T19:57:44Z TRACE agent::util::slot_reconciliation] reconcile - get_node_slots failed: Os { code: 13, kind: PermissionDenied, message: "Permission denied" }
[2020-10-27T19:57:44Z TRACE agent::util::slot_reconciliation] periodic_slot_reconciliation - iteration end
...
**REPEATS**
...
[2020-10-27T19:58:50Z TRACE agent::util::config_action] handle_config - something happened to a configuration
[2020-10-27T19:58:50Z INFO  agent::util::config_action] handle_config - added DevCapConfig nessie
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: hyper::Error(Connect, "invalid URL, scheme is not http")', agent/src/util/config_action.rs:146:64
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

The Nessie URL is https://www.lochness.co.uk/livecam/img/lochness.jpg

**Additional context**

**Comment by DazWilkin** [comment]
Ah, perhaps this is addressed by: https://github.com/deislabs/akri/pull/71#issuecomment-717475838

**Comment by bfjelds** [comment]
> Ah, perhaps this is addressed by: [#71 (comment)](https://github.com/deislabs/akri/pull/71#issuecomment-717475838)

for me it was a combination of using reqwest in the code (https://github.com/bfjelds/akri-1/blob/0f6bebfcbc81dfb3b48112d4862d32cdcf3111ef/build/containers/Dockerfile.nessie) and adding cert info to the agent/nessie containers (https://github.com/deislabs/akri/pull/71#issuecomment-717475838)

**Comment by DazWilkin** [comment]
Yes, I have it working ... thanks for all the help.

Yes, it's kinda involved :-)
